pkg/cidata/cidata.TEMPLATE.d/boot.sh

```diff
@@ -12,6 +12,14 @@ WARNING() {
 # shellcheck disable=SC2163
 while read -r line; do export "$line"; done <"${LIMA_CIDATA_MNT}"/lima.env
 
+sed -i '/#LIMA-START/,/#LIMA-END/d' /etc/environment
+cat "${LIMA_CIDATA_MNT}/lima.environment" >>/etc/environment
+
+# shellcheck disable=SC2163
+while read -r line; do
+	[ "$(expr "$line" : '^#.*')" -eq 0 ] && export "$line"
```

Comment:
The `^` at the beginning of the regular expression is implied:

```console
$ expr " #FOO" : "#"
0
$ expr "#FOO" : "#"
1
```

On some `expr` versions (e.g. `busybox` as used by Alpine) it generates some noise on stderr:

```console
$ expr "#FOO" : "^#"
expr: warning: '^#': using '^' as the first character
of a basic regular expression is not portable; it is ignored
1
```

So the current version works, but I would prefer to avoid the warning and simplify as:

```suggestion
	[ "$(expr "$line" : '#')" -eq 0 ] && export "$line"
```

Otherwise, it works exactly as I would have hoped, and I would have approved, except for this little nit.

Thank you for staying with my changing opinions on how this should be implemented!

Reply:
Oh, almost forgot: could you please squash the previous commits that have effectively been reverted? And also clean up the resulting commit message. Thank you!